src/crewai/agent.py

```diff
@@ -394,7 +394,7 @@ def _render_text_description_and_args(self, tools: List[Any]) -> str:
         """
         tool_strings = []
         for tool in tools:
-            args_schema = str(tool.args)
+            args_schema = str(tool.model_fields)
```

Comment:
This change will break the "Tool Arguments:" being passed to the model. For example, this is what gets passed for delegation tools:
```
Tool Name: Delegate work to coworker(task: str, context: str, coworker: Optional[str] = None, **kwargs)
Tool Description: Delegate a specific task to one of the following coworkers: flight_agent, hotel_agent, User
The input to this tool should be the coworker, the task you want them to do, and ALL necessary context to execute the task, they know nothing about the task, so share absolute everything you know, don\'t reference things but instead explain them.
Tool Arguments: {\'name\': FieldInfo(annotation=str, required=True), \'description\': FieldInfo(annotation=str, required=False, default=\'\'), \'args_schema\': FieldInfo(annotation=type[BaseModel], required=True, description=\'The tool schema.\', metadata=[SkipValidation()]), \'return_direct\': FieldInfo(annotation=bool, required=False, default=False), \'verbose\': FieldInfo(annotation=bool, required=False, default=False), \'callbacks\': FieldInfo(annotation=Union[list[BaseCallbackHandler], BaseCallbackManager, NoneType], required=False, default=None, exclude=True), \'callback_manager\': FieldInfo(annotation=Union[BaseCallbackManager, NoneType], required=False, default=None, description=\'.. deprecated:: 0.1.7 Use :meth:`~callbacks` instead.\
\
Callback manager to add to the run trace.\', exclude=True), \'tags\': FieldInfo(annotation=Union[list[str], NoneType], required=False, default=None), \'metadata\': FieldInfo(annotation=Union[dict[str, Any], NoneType], required=False, default=None), \'handle_tool_error\': FieldInfo(annotation=Union[bool, str, Callable[list, str], NoneType], required=False, default=False), \'handle_validation_error\': FieldInfo(annotation=Union[bool, str, Callable[list, str], NoneType], required=False, default=False), \'response_format\': FieldInfo(annotation=Literal[\'content\', \'content_and_artifact\'], required=False, default=\'content\'), \'func\': FieldInfo(annotation=Union[Callable[..., Any], NoneType], required=False, default=None), \'coroutine\': FieldInfo(annotation=Union[Callable[..., Awaitable[Any]], NoneType], required=False, default=None)}
Tool Name: Ask question to coworker(question: str, context: str, coworker: Optional[str] = None, **kwargs)
Tool Description: Ask a specific question to one of the following coworkers: flight_agent, hotel_agent, User
The input to this tool should be the coworker, the question you have for them, and ALL necessary context to ask the question properly, they know nothing about the question, so share absolute everything you know, don\'t reference things but instead explain them.
Tool Arguments: {\'name\': FieldInfo(annotation=str, required=True), \'description\': FieldInfo(annotation=str, required=False, default=\'\'), \'args_schema\': FieldInfo(annotation=type[BaseModel], required=True, description=\'The tool schema.\', metadata=[SkipValidation()]), \'return_direct\': FieldInfo(annotation=bool, required=False, default=False), \'verbose\': FieldInfo(annotation=bool, required=False, default=False), \'callbacks\': FieldInfo(annotation=Union[list[BaseCallbackHandler], BaseCallbackManager, NoneType], required=False, default=None, exclude=True), \'callback_manager\': FieldInfo(annotation=Union[BaseCallbackManager, NoneType], required=False, default=None, description=\'.. deprecated:: 0.1.7 Use :meth:`~callbacks` instead.\
\
Callback manager to add to the run trace.\', exclude=True), \'tags\': FieldInfo(annotation=Union[list[str], NoneType], required=False, default=None), \'metadata\': FieldInfo(annotation=Union[dict[str, Any], NoneType], required=False, default=None), \'handle_tool_error\': FieldInfo(annotation=Union[bool, str, Callable[list, str], NoneType], required=False, default=False), \'handle_validation_error\': FieldInfo(annotation=Union[bool, str, Callable[list, str], NoneType], required=False, default=False), \'response_format\': FieldInfo(annotation=Literal[\'content\', \'content_and_artifact\'], required=False, default=\'content\'), \'func\': FieldInfo(annotation=Union[Callable[..., Any], NoneType], required=False, default=None), \'coroutine\': FieldInfo(annotation=Union[Callable[..., Awaitable[Any]], NoneType], required=False, default=None)}
```